gitapp/lib/HomePage/homepage.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:gitapp/constants/constants.dart';
+import 'package:gitapp/user_details/user_details.dart';
+import 'package:gitapp/users_list/User_card.dart';
+
+import 'package:gitapp/users_list/users_list.dart';
+
+class HomePage extends StatefulWidget {
+  HomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  HomePageState createState() => HomePageState();
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  List<UserCard> _card = <UserCard>[];
+  final UsersList users = new UsersList();
+  String name = "";
+  final TextEditingController _textController = new TextEditingController();
+  bool searching = false;
+
+  Future _getUsers(String text) async {
+    _card.clear();
+
+    if (text != "") {
+      name = text;
+      _textController.clear();
+
+      //users.fillSearchedUsers(text);
+      await users.fetchUsers(text);
+
+      if (users.searchedlist.length != 0) {
+        users.searchedlist.forEach((element) {
```

Comment:
Also we should not use foreach.
If We want to transform an collection of items into a collection of items of different type, we should use map. method. 

Please refer to the following link.

https://medium.com/flutter-community/useful-list-methods-in-dart-6e173cac803d

Reply:
Solved